R/timeseriessummary-data.R

```diff
@@ -388,4 +389,86 @@ parseTSSGapTolerances <- function(reportData, timezone){
   }
   
   return(gapTolerances)
-}
\ No newline at end of file
+}
+
+#' Format TSS Correction Parameters
+#' @description formats the parameters so they are prettier to print
+#' @param corrections the corrections for the report
+#' @param timezone the timezone to parse data into
+#' @return corrections with formattedParameters field to print in report
+formatCorrectionParameters <- function(corrections, timezone) {
+  
+  corrections[["formattedParameters"]] <- ""
+  
+  if(!isEmptyOrBlank(corrections) && !isEmptyVar(corrections)) {
+    
+    for (i in seq(nrow(corrections))) {
```

Comment:
bleh can't figure out how to vectorize this because the format is so funky. 

However, I think we should get rid of all those if-else statments and separate this out into many smaller, formatting functions. Then we can use a switch statement. This would be the only thing inside the for loop.

```
corrections[i, "formattedParameters"] <- 
        switch(corrections[i, 'type'],
               "Offset" = formatCorrectionsParamOffset(corrections[i, 'parameters'], timezone),
               "Drift" = formatCorrectionsParamDrift(corrections[i, 'parameters'], timezone),
               "SinglePoint" = formatCorrectionsParamSinglePoint(corrections[i, 'parameters'], timezone),
               "USGSMultiPoint" = formatCorrectionsParamUSGSMultiPoint(corrections[i, 'parameters'], timezone),
               "AdjustableTrim" = formatCorrectionsParamAdjustableTrim(corrections[i, 'parameters'], timezone),
               "FillGaps" = formatCorrectionsParamFillGaps(corrections[i, 'parameters'], timezone),
               "Deviation" = formatCorrectionsParamDeviation(corrections[i, 'parameters'], timezone),
               corrections[i, 'type']) #anything that doesn't match the previous words will get this as output
```